### PR TITLE
feat: add auto generated jobs

### PR DESCRIPTION
**Pull request by Henrrypg**
This PR adds the functionality of autogenerated init jobs and migration jobs. 

Both are working fine. 

NOTE: The migration can't be tested with any filled database due to MySQL 8 is required and we are currently working with MySQL 5.7. The migration to MySQL 8 should be another task.

### CODE REVIEW

**Comment by MoisesGSalas** [comment]
> NOTE: The migration can't be tested with any filled database due to MySQL 8 is required and we are currently working with MySQL 5.7. The migration to MySQL 8 should be another task.

Maybe we could test this by using a MySQL8 DB from the start using an external mysql instead of the one managed by tutor.

**Comment by Henrrypg** [comment]
@MoisesGSalas i was testing and i ran into de following error:

Context: Running init jobs in maple installation with external mysql 8 db.

Error: 

`django.db.utils.OperationalError: (1059, "Identifier name 'blackboard_blackboardlearnerassessmentdatatransmissionaudit_chk_1' is too long")`

This is due to identifier limit in MySQL 8. https://dev.mysql.com/doc/refman/8.0/en/identifier-length.html 

Any thoughts about a way to achieve that?

**Comment by MoisesGSalas** [comment]
That seems to be fixed on https://github.com/openedx/edx-enterprise/releases/tag/3.40.0,

nutmeg uses [3.42](https://github.com/openedx/edx-platform/blob/open-release/nutmeg.1/requirements/edx/base.txt#L479) and maple uses [3.30](https://github.com/openedx/edx-platform/blob/open-release/maple.3/requirements/edx/base.txt#L435).

I just test from nutmeg onwards

File: drydock/templates/drydock/k8s/jobs.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        command:
+          - /bin/sh
+          - -c
+          - -e
+        args:
+        - |
+          {%- if DRYDOCK_MIGRATE_FROM == 13 %}
+          ./manage.py lms backpopulate_user_tours
+          {%- endif %}
+          {%- if DRYDOCK_MIGRATE_FROM in (13,14) %}
```
Comment (marbonilla) [inline drydock/templates/drydock/k8s/jobs.yml:35]: to migrate from versions higher than 13 and 14 of open edx, is there any additional execution required?

Comment (Henrrypg) [reply drydock/templates/drydock/k8s/jobs.yml:35]: No sr. you can look at which commands we are using in [tutor upgrade command](https://github.com/overhangio/tutor/blob/master/tutor/commands/upgrade/k8s.py#L87)

**Comment by Henrrypg** [comment]
@MoisesGSalas It seems to work fine upgrading  from nutmeg to quince, i put a note with the warning of the needed of have MySQL 8.0 before the migration.

File: README.rst
Comment (MoisesGSalas) [inline README.rst:1]: I've just noticed that this file is `.rst` but has some markdown formatting. Can we either transform to markdown or fix the current formatting?

File: README.rst [hunk 2]
```diff
@@ -51,19 +51,65 @@ The following configuration options are available:
...
+- `Upgrade Jobs`: When **DRYDOCK_MIGRATE_FROM** is set, over the Sync Wave 50
+- `CMS and LMS Deployments`: When **DRYDOCK_POD_LIFECYCLE** is active, over the Sync Wave 100
+- `Debug Resources`: When **DRYDOCK_DEBUG** active, over the Sync Wave 100
+- `Horizontal Pod Autoscalers`: When active, over the Sync Wave 150
+
+Workaround to upgrade from Maple to Palm or later
+-------------------------------------------------
+
+.. note::
+    To upgrade to Quince version, you must first upgrade to MySQL 8.0. This is because the Quince version uses Django 3.2, which is not compatible with MySQL 5.7.
```
Comment (MoisesGSalas) [inline README.rst:96]:
```suggestion
    Quince uses Django 4.2 which only supports MySQL 8 or higher. You must upgrade your version of MySQL prior to performing the upgrade.
```

File: README.rst
Comment (Henrrypg) [reply README.rst:1]: ok, i will fix it

File: README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+[Tutor filter](https://docs.tutor.edly.io/reference/api/hooks/filters.html) **SYNC_WAVES_ORDER** was added to allow define [ArgoCD Sync Waves](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/) order and apply to the kubernetes resources through **get_sync_waves_for_resource** function.
+
+We are defined by defult the following order:
+
+- `All kubernetes resources` (except the ones that are defined in the next waves)
+- `Initialization Jobs`
+- `Upgrade Jobs`: When **DRYDOCK_MIGRATE_FROM** is set, over the Sync Wave 50
+- `CMS and LMS Deployments`: When **DRYDOCK_POD_LIFECYCLE** is active, over the Sync Wave 100
+- `Debug Resources`: When **DRYDOCK_DEBUG** active, over the Sync Wave 100
+- `Horizontal Pod Autoscalers`: When active, over the Sync Wave 150
```
Comment (MoisesGSalas) [inline README.md:88]:
```suggestion
1. `All kubernetes resources` (except the ones that are defined in the next waves)
2. `Initialization Jobs`
3. `Upgrade Jobs`: When **DRYDOCK_MIGRATE_FROM** is set, over the Sync Wave 50
4. `CMS and LMS Deployments`: When **DRYDOCK_POD_LIFECYCLE** is active, over the Sync Wave 100
5. `Debug Resources`: When **DRYDOCK_DEBUG** active, over the Sync Wave 100
6. `Horizontal Pod Autoscalers`: When active, over the Sync Wave 150
```

If they are ordered it's better to use an ordered list.

**Review by MoisesGSalas — APPROVED** [review]

**Review by marbonilla — APPROVED** [review]

**Review by jfavellar90 — APPROVED** [review]

**Pull request merged**